sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/StreamExecution.scala

```diff
@@ -52,13 +55,28 @@ class StreamExecution(
   /** Minimum amount of time in between the start of each batch. */
   private val minBatchTime = 10
 
-  /** Tracks how much data we have processed from each input source. */
-  private[sql] val streamProgress = new StreamProgress
+  /**
+   * Tracks how much data we have processed and committed to the sink or state store from each
+   * input source.
+   */
+  private[sql] val committedOffsets = new StreamProgress
```

Comment:
We need some protection for accessing committedOffsets since StreamProgress is not thread-safe. How about providing a getter method that is protected by synchronized and returns a clone `StreamProgress`? Since this is only used in tests, we don't need to worry about the overhead of `clone`.